src/PolyType.SourceGenerator/Parser/Parser.Diagnostics.cs

```diff
@@ -110,4 +110,36 @@ public sealed partial class Parser
         category: "PolyType.SourceGenerator",
         defaultSeverity: DiagnosticSeverity.Error,
         isEnabledByDefault: true);
+
+    private static DiagnosticDescriptor AssociatedTypeInternal { get; } = new DiagnosticDescriptor(
+        id: "TS0014",
+        title: "Associated type is internal.",
+        messageFormat: "The associated type '{0}' is internal, which will prevent its activation from within a [GenerateShape] graph produced from another assembly. Associated types should generally be public.",
```

Comment:
That should hold true for pretty much anything within a `[GenerateShape]` type graph that is marked internal, but we generally don't warn about it. Perhaps it makes sense to only emit a diagnostic in cases where the source generator cannot access the associated type?

Reply:
Ok, yes I see `TypeNotAccessible` is there. I'll delete this and use that.